### PR TITLE
[#389] Fix checks of bucket settings

### LINKED ISSUE #389: PutObject, CopyObject, PutBucketVersioning are not working if a bucket hasn't a settings file
How I produced: 
1. Create a bucket 
2. Try to put a file
3. S3 GW returns an error `NoSuchKey`, which is returned by `GetBucketSettings` function

### PR DESCRIPTION
**Pull request by masterSplinter01**
Closes #389

### CODE REVIEW

File: api/handler/put.go [hunk 1]
```diff
@@ -206,15 +206,17 @@ func (h *handler) PutObjectHandler(w http.ResponseWriter, r *http.Request) {
...
 		h.logAndSendError(w, "could not get bucket settings", reqInfo, err)
 		return
 	}
 
-	params.Lock, err = formObjectLock(bktInfo, settings.LockConfiguration, r.Header)
-	if err != nil {
-		h.logAndSendError(w, "could not form object lock", reqInfo, err)
-		return
+	if settings != nil {
+		params.Lock, err = formObjectLock(bktInfo, settings.LockConfiguration, r.Header)
```
Comment (KirillovDenis) [inline api/handler/put.go:215]: Actually, we have to check `bktInfo.ObjectLockEnabled` (it's done in `formObjectLock` function)
```suggestion
	if settings == nil {
	        settings = &data.BucketSettings{}
	}
```

File: api/handler/versioning.go [hunk 2]
```diff
@@ -62,7 +65,7 @@ func (h *handler) GetBucketVersioningHandler(w http.ResponseWriter, r *http.Requ
 
 	settings, err := h.obj.GetBucketSettings(r.Context(), bktInfo)
 	if err != nil {
-		if errors.IsS3Error(err, errors.ErrNoSuchBucket) {
+		if errors.IsS3Error(err, errors.ErrNoSuchKey) {
```
Comment (KirillovDenis) [inline api/handler/versioning.go:68]: We still have to return `VersioningConfiguration` even though we haven't settings file. So this shouldn't be changed

File: api/handler/copy.go [hunk 3]
```diff
@@ -109,15 +109,17 @@ func (h *handler) CopyObjectHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	settings, err := h.obj.GetBucketSettings(r.Context(), dstBktInfo)
-	if err != nil {
+	if err != nil && !errors.IsS3Error(err, errors.ErrNoSuchKey) {
 		h.logAndSendError(w, "could not get bucket settings", reqInfo, err)
 		return
 	}
```
Comment (KirillovDenis) [inline api/handler/copy.go:115]: I think we can refactor `h.obj.GetBucketSettings` so it handle `errors.ErrNosuchKey`

Comment (masterSplinter01) [reply api/handler/copy.go:115]: Do you mean that `h.obj.GetBucketSettings` should return `nil, nil` if no settings file is found?

Comment (KirillovDenis) [reply api/handler/copy.go:115]: No, it should return `&data.BucketSettings{}, nil` if settings file isn't found

File: api/handler/put.go [hunk 4]
```diff
@@ -249,9 +249,7 @@ func (h *handler) PutObjectHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	if settings, err = h.obj.GetBucketSettings(r.Context(), bktInfo); err != nil {
-		h.log.Warn("couldn't get bucket versioning", zap.String("bucket name", reqInfo.BucketName), zap.Error(err))
-	} else if settings.VersioningEnabled {
+	if settings != nil && settings.VersioningEnabled {
```
[the remark this answers is not in the document]
Comment (KirillovDenis) [reply api/handler/put.go]: The same for `CompleteMultipartUploadHandler`

[the remark this answers is not in the document]
Comment (masterSplinter01) [reply api/handler/put.go]: What's wrong with `CompleteMultipartUploadHandler`? 
```
	if settings, err := h.obj.GetBucketSettings(r.Context(), bktInfo); err != nil {
		h.log.Warn("couldn't get bucket versioning", zap.String("bucket name", reqInfo.BucketName), zap.Error(err))
	} else if settings.VersioningEnabled {
		w.Header().Set(api.AmzVersionID, objInfo.Version())
	}
}
```

[the remark this answers is not in the document]
Comment (KirillovDenis) [reply api/handler/put.go]: Sorry, confused with https://github.com/nspcc-dev/neofs-s3-gw/pull/386

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**